feat: copilot list for automatic combat

ninekirin opened the pull request:
implements automatic continuous combat for stages in the same area

仿照 WpfGui 实现自动战斗列表功能

<img width="1004" alt="image" src="https://github.com/user-attachments/assets/1f7b4be9-0404-42f2-be5f-08ddb65c32a4" />

**主要变更：**
- 战斗列表功能（见 203ca9b）
  - 支持自动导航、突袭，基本上是照着 WpfGui 那边来实现的，对 View 的改动比较大，原有的 CopilotContent 拆分成了 CopilotContent + CopilotList + RegularCopilotList
  - CopilotView 页增加添加到战斗列表的选项
  - 追踪执行 copilot 任务的回调主要在 MaaMessage 中实现
    - （一个想法）MAACore 在 TaskChainCompleted 后的回调中有 taskid，未来可以对已启用的 task 和添加到 MAACore 中的 task 进行 mapping，更精确的判断任务状态，目前考虑到战斗列表是顺序执行所以没有实现
- 原有的 copilots 列表并不是在程序启动后预载，而是列表页面加载时从 copilot 和 recording 目录读取，没有持久化设计，导致每次切换都会重新读取目录；我做了一些变更，将大部分变量移动到 ViewModel（见 203ca9b）
  - 里面有些地方我不太会处理双向数据绑定（
- 新增一个 MapHelper 用于处理 stage_name 和 navigate_name 之间的关联关系（见 dd321d9）
- 给部分 Models 增加参数（见 5c96b19）
- 增加了 downloads 目录的只读权限，解决 `.fileImporter` 读取不了沙盒外文件的问题（我不确定这么做是否合适，亦或是使用startAccessingSecurityScopedResource更合适一些）（见 3e22c3b）
- 给部分 Xcode 中不能预览的文件增加 EnvironmentObject（见 adc1745, 364a969, 369b676） 
- Typo（见95dc024）

**目前已知会影响战斗列表的问题：**
- Mac 下的跳过剧情有概率不工作，导致中途卡在跳过剧情页

我之前没怎么接触过 Swift，实现里难免有不太到位的地方，如有不当之处还请大佬不吝指正（

ninekirin committed:
fix(preview): change deprecated SSS to OF-1

ninekirin committed:
fix(preview): add MAAViewModel for preview

ninekirin committed:
chore: code format & cleanup

ninekirin committed:
fix: typo in TaskDetail

fix typo in TaskDetail (misspelled ViewDetailTabButton)

ninekirin committed:
fix: no permission reading user downloads

ninekirin committed:
feat: copilot list models preparation

ninekirin committed:
feat: MapHelper for copilot list navigation

用于寻找地图的 stage_name (例如 act42side_10) 对应的 navigate_name (MT-10)

ninekirin committed:
feat: update all views for the copilot list

ninekirin committed:
fix(preview): add MAAViewModel for UtilityContent

ChestnutLUO commented:
UX角度几个建议：
页面默认显示单个作战（原先的界面）。
copilotlist 那个切换代码不要用.fill symbol切换，做成symbol着色感觉会直观一点。

ninekirin committed:
fix: select new file after switching back

- fixed the newly downloaded copilot file can not be selected automatically after switching back from the copilot list after downloading
- added onSubmit to accepts keyboard Enter input for prtsCode

- 修复了在战斗列表页面下载作业自动切回来之后不会选中新下载的作业的问题
- 神秘代码的 TextField 绑定 onSubmit，接受键盘 Enter 输入

ninekirin committed:
fix: remove MAAUseCopilotList persistence

removed MAAUseCopilotList persistence and show RegularCopilotList by default on startup

移除了 MAAUseCopilotList 的持久化，应用启动默认展示 RegularCopilotList

ninekirin committed:
feat: change toggle button style

Ir1d commented on MeoAsstMac/Model/MAAViewModel.swift:
```diff
@@ -454,35 +471,283 @@ extension MAAViewModel {
+            name: name,
+            is_raid: is_raid,
+            need_navigate: true,
+            navigate_name: navigate_name,
```
我发现这个逗号会导致编译失败 🥺

ninekirin committed:
chore: remove redundant comma

Ir1d commented:
我在尝试 "[矢量突破自动战斗循环第一层](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/12726)"，但好像这个 pr 还不太支持矢量突破，想请教一下这个矢量突破的导航需要改哪一部分 code 呢？

ChestnutLUO commented:
> 我在尝试 "[矢量突破自动战斗循环第一层](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/12726)"，但好像这个 pr 还不太支持矢量突破，想请教一下这个矢量突破的导航需要改哪一部分 code 呢？

<img width="635" alt="image" src="https://github.com/user-attachments/assets/27d38dfc-b315-4787-b4b1-552ae07e6f6d" />
这样吗？

ChestnutLUO commented:
> 我在尝试 "[矢量突破自动战斗循环第一层](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/12726)"，但好像这个 pr 还不太支持矢量突破，想请教一下这个矢量突破的导航需要改哪一部分 code 呢？

具体哪里不支持？
显示“不支持的关卡”的话你要重新编译一下maa core

Ir1d commented:
maa core 是昨天编译的 main 应该是够新的，我是从这个界面启动 
<img width="1494" alt="image" src="https://github.com/user-attachments/assets/4acdc3ca-22aa-4f71-b8f2-b0c893f1d67d" />
然后设置列表
<img width="287" alt="image" src="https://github.com/user-attachments/assets/fdad5a1a-e5fe-4370-9f5e-24f256127056" />

此时如果启动列表，log 会停在 “开始任务：copilot”，并不进入关卡；而如果不在战斗列表页而是默认 copilot view 页面点启动是可以进入关卡正常完成一轮行动的 🥺  我的操作哪里有问题吗？如果没有的话，如何进一步 debug 毕竟好？

ChestnutLUO commented:
感觉应该去问maa core的开发🤔

Ir1d commented:
感谢！on a side note.. 我在这里 https://github.com/MaaAssistantArknights/MaaMacGui/blob/ab001ca1847be4ed8a0dae96783a2ff549682065/MeoAsstMac/Views/CopilotView.swift#L49-L74 把 https://github.com/MaaAssistantArknights/MaaMacGui/blob/ab001ca1847be4ed8a0dae96783a2ff549682065/MeoAsstMac/Views/CopilotView.swift#L81-L85 加进去之后是可以实现矢量突破自动战斗循环第一层的

ninekirin commented:
> 感谢！on a side note.. 我在这里
> 
> https://github.com/MaaAssistantArknights/MaaMacGui/blob/ab001ca1847be4ed8a0dae96783a2ff549682065/MeoAsstMac/Views/CopilotView.swift#L49-L74
> 
> 把
> https://github.com/MaaAssistantArknights/MaaMacGui/blob/ab001ca1847be4ed8a0dae96783a2ff549682065/MeoAsstMac/Views/CopilotView.swift#L81-L85
> 
> 加进去之后是可以实现矢量突破自动战斗循环第一层的，只是不知道为啥战斗列表不支持

因为战斗列表给 MAACore 传的参数里面有一条 `need_navigate: true`，因此开始前 MAA 会试图寻找名为 "VEC-01" 的关卡，尚未在画面中识别到这个关卡的话会一直寻找，看日志就可以发现状态卡在了 FullStageNavigation。因此战斗列表只支持日常战斗，不支持保全和矢量突破这种特殊模式~

还是等 hg 大佬重构常规自动战斗之后加入循环次数的支持吧~

Ir1d commented:
哦哦 原来如此 感谢！ 🥺 没事我上一条那样改一下已经可以循环 vec-01 了（躺平

ninekirin committed:
Merge branch 'master' into feat/copilot-list-pr

ninekirin committed:
feat: implement user_additional param

ninekirin commented:
我在自动战斗配置页增加了追加自定干员的调用（见 8e4c88a），和 WpfGui 那边对齐了（

<img width="1004" alt="image" src="https://github.com/user-attachments/assets/6c4480eb-e629-43d0-8ecb-020a2e8bca8e" />

ninekirin committed:
fix: i18n

ninekirin committed:
Merge branch 'MaaAssistantArknights:master' into feat/copilot-list-pr

ninekirin committed:
Merge branch 'MaaAssistantArknights:master' into feat/copilot-list-pr

ninekirin committed:
Merge branch 'master' into feat/copilot-list-pr

ninekirin committed:
Merge branch 'MaaAssistantArknights:master' into feat/copilot-list-pr

ChestnutLUO commented:
https://github.com/MaaAssistantArknights/MaaMacGui/commit/2e48e30fdb79aaac6d98846b20f64f703b0f27aa

现已支持战斗列表功能
